core/src/main/java/org/apache/iceberg/puffin/Blob.java

```diff
@@ -30,29 +30,23 @@
 public final class Blob {
   private final String type;
   private final List<Integer> inputFields;
-  private final long snapshotId;
-  private final long sequenceNumber;
   private final ByteBuffer blobData;
   private final PuffinCompressionCodec requestedCompression;
   private final Map<String, String> properties;
 
-  public Blob(
-      String type, List<Integer> inputFields, long snapshotId, long sequenceNumber,
-      ByteBuffer blobData) {
-    this(type, inputFields, snapshotId, sequenceNumber, blobData, null, ImmutableMap.of());
+  public Blob(String type, List<Integer> inputFields, ByteBuffer blobData) {
+    this(type, inputFields, blobData, null, ImmutableMap.of());
   }
 
   public Blob(
-      String type, List<Integer> inputFields, long snapshotId, long sequenceNumber,
-      ByteBuffer blobData, @Nullable PuffinCompressionCodec requestedCompression, Map<String, String> properties) {
+      String type, List<Integer> inputFields, ByteBuffer blobData,
+      @Nullable PuffinCompressionCodec requestedCompression, Map<String, String> properties) {
     Preconditions.checkNotNull(type, "type is null");
     Preconditions.checkNotNull(inputFields, "inputFields is null");
     Preconditions.checkNotNull(blobData, "blobData is null");
     Preconditions.checkNotNull(properties, "properties is null");
     this.type = type;
     this.inputFields = ImmutableList.copyOf(inputFields);
-    this.snapshotId = snapshotId;
-    this.sequenceNumber = sequenceNumber;
```

Comment:
I think the snapshot and sequence number for the blob should be retained. That allows preserving blobs that were not updated.

Reply:
It is related to the metadata added here, but unnecessary for the purpose of this PR. I think it needs more discussion and I'd like to focus on making progress on the main content of the PR.

Right now, I need to think about this more. The rationale for these fields is unchanged, although we could reconsider it by making the search for relevant metrics more complicated.